cpp/src/fil/infer.cu

```diff
@@ -344,7 +327,7 @@ struct tree_aggregator_t<NITEMS, CATEGORICAL_LEAF> {
     for (int c = threadIdx.x; c < num_classes; c += FIL_TPB * NITEMS)
 #pragma unroll
       for (int item = 0; item < NITEMS; ++item) votes[c * NITEMS + item] = 0;
-    //__syncthreads(); // happening outside already
+    //__syncthreads() is done in infer_k
```

Comment:
`// __syncthreads() is performed in infer_k` (extra space between `//` and `__syncthreads()`, to avoid confusion with commented-out code).